### ISSUE TITLE
[Feature Request] DatePickerThemeData to custom style of disabled dates

### ISSUE THREAD

**Issue by jasonkang14** [flutter/flutter#128378]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Use case

I am using `showDatePicker()` to have my users use a calendar to select a date. 
I would like my `enabled` dates which meet the conditions of `selectableDayPredicate` to look different from `disabled` dates. so I looked at the source code of `calendar_date_picker.dart` and changed `_DayPickerState` like below

```dart
...
final Color? dayForegroundColor = resolve<Color?>(
            (DatePickerThemeData? theme) => isToday ? theme?.todayForegroundColor : theme?.dayForegroundColor, states);
final Color? dayBackgroundColor = resolve<Color?>(
            (DatePickerThemeData? theme) => isToday ? theme?.todayBackgroundColor : theme?.dayBackgroundColor, states);
final Color? disabledColor = resolve<Color?>(
            (DatePickerThemeData? theme) => isDisabled ? theme?.dayOverlayColor : theme?.dayOverlayColor, states);
...
Widget dayWidget = Container(
          decoration: decoration,
          child: Center(
            child: Text(
              localizations.formatDecimal(day),
              style: dayStyle?.apply(
                color: isDisabled ? disabledColor : dayForegroundColor,
              ),
            ),
          ),
        );

```

### Proposal

<img width="361" alt="Screenshot 2023-06-07 at 1 37 50 PM" src="https://github.com/flutter/flutter/assets/45306565/04a7d375-1374-405e-9863-40f20154776a">

the `CalendarDatePicker` now looks like the screenshot above, but I would like to have the disabled dates different `fontWeight` and `fontSize` from enabled dates. would you guys be willing to accept a PR if I were to add this feature?

**Comment by huycozy** [comment]
This is somewhat similar to https://github.com/flutter/flutter/issues/93707 but this proposal is more specific on `DatePickerThemeData`'s disabled date style.

**Comment by jasonkang14** [comment]
yes I was thinking maybe something like `disabledDayStyle` could be used along with `dayStyle`. I understand that `dayStyle` overrides the default text style used for each individual day label in the grid of the date picker, including the disabled days. so maybe another property could be used?

**Comment by HansMuller** [comment]
I'm probably missing some nuance here but: can't the day{Foreground,BackgroundOverlay} `MaterialState<Color?>` properties resolve to whatever colors you want for disabled/enabled dates?

**Comment by jasonkang14** [comment]
those two properties are applied to both enabled and disabled dates. I was asking if we could add add a property which would allow users to apply different styles to enabled dates and disabled dates

**Comment by HansMuller** [comment]
Maybe I'm missing something ... why can't the resolve methods for the `MaterialStateProperty<Color>` values you specify for those properties depend on the enabled/disabled state?

**Comment by jasonkang14** [comment]
oh i may not have been clear. the attached code above is my forked version. the source code actually looks like this

```dart
Widget dayWidget = Container(
          decoration: decoration,
          child: Center(
            child: Text(localizations.formatDecimal(day), style: dayStyle?.apply(color: dayForegroundColor)),
          ),
        );

        if (isDisabled) {
          dayWidget = ExcludeSemantics(
            child: dayWidget,
          );
        }
```

there is no way to apply different styles to enabled/disabled dates

**Comment by jasonkang14** [comment]
are you guys still interested in this if it has not been implemented?
